[Telfer] iP

telferang opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

telferang committed:
Rename Duke to Nova, Greet, Exit

telferang committed:
Merge remote-tracking branch 'origin/add-gradle-support'

telferang committed:
Add Echo feature

For level-1

telferang committed:
Add Add and List feature

For level-2

telferang committed:
Add mark as done features to Nova

Task class: Manage individual tasks, with status tracking
Nova.java: Modify code to handle Task class and handle invalid input

telferang committed:
Follow the coding standard

telferang committed:
Add Todo, Event and Deadline class

Todo, Event and Deadline class allow users to specify the type of task
to add

telferang committed:
Add TextUiTesting

Holy-An commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,214 @@
+
+    private static void handleMarkAndUnmarkCase(String[] words, Task[] tasks) {
+        int taskIndex = validateIndex(words);
+        if (taskIndex == -1) return;
```
According to the coding standards, The if-else class of statements would have the following form:
if (condition) {
    statements;
}

Perhaps you could refactor your code? From
if (taskIndex == -1) return;

to

if (taskIndex == -1) {
    return;
}

wongwh2002 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,214 @@
+    private static void handleListCase(Task[] tasks) {
+        System.out.println(SEPARATOR);
+        for (int i = 1; i <= Task.getNumberOfTasks(); i++) {
+            System.out.println("     " + i + "." + tasks[i - 1].getTaskInfo());
```
Perhaps use "/t" instead of "(4spaces)"?

wongwh2002 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,214 @@
+    }
+
+    private static void handleTodoCase(String[] words, Task[] tasks) {
+        if (words.length != 2) {
```
Perhaps this can be extracted out instead of using magic numbers?

wongwh2002 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,214 @@
+                return;
+            }
+        }
+    }
```
Perhaps you can use a flag to break out of while loop instead of using while true?

wongwh2002 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,214 @@
+        splitInput[2] = input[1];
+        return splitInput;
+    }
+}
```
Coding quality is generally good, use of proper indentation and naming conventions.

Holy-An replied:
Your methods are clear and well-defined. Perhaps you could add more javadoc comments to elaborate your methods further?

Holy-An commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,214 @@
+
+        case "event":
+            handleEventCase(inputs, tasks);
+            break;
```
Really clear and well-defined test cases!

Holy-An commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,214 @@
+
+    private static final int MAX_TASKS = 100;
+    public static final String SEPARATOR = "    ___________________________________________________________________";
+    public static final String NEW_LINE = "\n     ";
```
Very clear definition of constants. Well done!

Holy-An reviewed (commented):
Overall, a fantastic job. Your code is clear and well-defined.

telferang committed:
Improve code quality

Introduced CommandHandler, InputValidator, MessageDisplay, TaskManager
class for better code quality

telferang committed:
Add invalid input error handling

telferang committed:
Merge branch 'branch-Level-5'

telferang committed:
Add InsufficientSpaceException

InsufficientSpaceException checks if user input more tasks than MAX_TASK

telferang committed:
Move all code into nova package

telferang committed:
Move all codes into nova package

telferang committed:
Move exception class into exception package

telferang committed:
Move all task class into task package

telferang committed:
Merge branch 'branch-A-Packages'

telferang committed:
Add delete function and Java Collections classes

telferang committed:
Add Save function

telferang committed:
Merge branch 'branch-Level-6'

telferang committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/nova/CommandHandler.java
#	src/main/java/nova/TaskManager.java
#	text-ui-test/runtest.bat

telferang committed:
Add JAR

telferang committed:
Add more OOP code

Add Command class and various Command sub classes
Rename the class according to the requirements

telferang committed:
Rename CommandHandler to Parser

telferang committed:
Change the date type from String to LocalDate in Deadline and Event

telferang committed:
Add optional date parameter to list command

Allow user to list Event and Deadline the is on the selected date

telferang committed:
Add Find feature to search for task containing specific phrase

telferang committed:
Add JavaDoc to all the classes

telferang committed:
Merge pull request #3 from telferang/branch-A-JavaDoc

Branch a java doc

telferang committed:
Merge branch 'master' of https://github.com/telferang/ip into branch-A-JavaDoc

telferang committed:
Merge remote-tracking branch 'origin/master' into branch-Level-9

# Conflicts:
#	src/main/java/nova/task/Task.java

telferang committed:
Merge pull request #2 from telferang/branch-Level-9

Branch level 9

telferang committed:
Merge branch 'master' of https://github.com/telferang/ip into branch-Level-9

telferang committed:
Merge remote-tracking branch 'origin/master' into branch-Level-8

# Conflicts:
#	data/nova.txt
#	src/main/java/nova/TaskList.java
#	src/main/java/nova/command/DeadlineCommand.java
#	src/main/java/nova/command/EventCommand.java
#	src/main/java/nova/command/ListCommand.java
#	src/main/java/nova/task/Deadline.java
#	src/main/java/nova/task/Event.java
#	src/main/java/nova/task/Task.java

telferang committed:
Merge pull request #1 from telferang/branch-Level-8

Branch level 8

telferang committed:
Merge branch 'branch-A-JavaDoc'

telferang committed:
Merge branch 'branch-Level-9'

telferang committed:
Clean the code

okkhoy commented on src/main/java/nova/Nova.java:
```diff
@@ -0,0 +1,49 @@
+     * It displays the welcome message, initializes the task manager and parser,
+     * loads the task storage, and enters a loop to handle user input.
+     *
+     * @param args Command-line arguments (not used).
```
Header comment doesn't follow the standard we specify

okkhoy commented on src/main/java/nova/Nova.java:
```diff
@@ -0,0 +1,49 @@
+        // Continuously reads and processes user input
+        while (true) {
+            String[] userInput = getUserInput(sc);
+            if (parser.handleInput(userInput)) {
```
it is probably unwise to return boolean in a `handleInput` method.
if that is indeed the usecase, the method should read boolean

okkhoy commented on src/main/java/nova/Nova.java:
```diff
@@ -0,0 +1,49 @@
+    private static String[] getUserInput(Scanner sc) {
+        if (!sc.hasNextLine()) {
+            return new String[]{"bye"}; // Default command to exit the program
+        }
```
could this be handled using an exception instead?

okkhoy commented on src/main/java/nova/Parser.java:
```diff
@@ -0,0 +1,74 @@
+package nova;
+
+import nova.command.*;
```
don't use wildcard imports

okkhoy commented on src/main/java/nova/Parser.java:
```diff
@@ -0,0 +1,74 @@
+            ListCommand.execute(inputs, taskManager);
+            break;
+
+        case MarkAndUnmarkCommand.COMMAND_WORD1:
```
should use fallthrough comment

okkhoy commented on src/main/java/nova/Storage.java:
```diff
@@ -0,0 +1,120 @@
+                    continue; // Skip empty lines
+                }
+
+                String[] line = nextLine.split(" \\| ");
```
SLAP well?

okkhoy commented on src/main/java/nova/command/Command.java:
```diff
@@ -0,0 +1,8 @@
+ * Represents a base class for all commands in the application.
+ * This class can be extended to implement specific command functionality.
+ */
+public class Command {
```
make it abstract?

okkhoy commented on src/main/java/nova/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,67 @@
+        if (inputs.length != 2) {
+            throw new InvalidInputException("No description entered.");
+        }
+        if (!inputs[1].contains("/by")) {
```
try SLAP here

okkhoy commented on src/main/java/nova/command/MarkAndUnmarkCommand.java:
```diff
@@ -0,0 +1,71 @@
+ * Represents a command to mark or unmark tasks in the task manager.
+ * This command allows users to update the completion status of tasks.
+ */
+public class MarkAndUnmarkCommand extends Command {
```
maybe creating 2 classes is better here?

okkhoy reviewed (commented):
In general, the code is well-written. There are a few specific instances that can be improved. Please take a look at the comments.

telferang committed:
Add user guide on READ.md

telferang committed:
Create JAR release

telferang committed:
Fix javadoc and README

telferang committed:
Change javadoc comment

telferang committed:
Update README by fixing typos
